diff --git a/lib/utils/prettify-metadata.js b/lib/utils/prettify-metadata.js
index 72483b14a05813ee6d67adeb6268afb193f0bc17..29bf3d0afda41c6a83899addf1092674288bb039 100644
--- a/lib/utils/prettify-metadata.js
+++ b/lib/utils/prettify-metadata.js
@@ -62,7 +62,7 @@ function prettifyMetadata ({ log, context }) {
       ? prettifiers.caller(log.caller, 'caller', log, { colors: colorizer.colors })
       : log.caller
 
-    line += `${line === '' ? '' : ' '}<${prettyCaller}>`
+    line += `${line === '' ? '' : ' '}(${prettyCaller})`
   }
 
   if (line === '') {
